### PR TITLE
Fix and improve 'west sign'

### PR DESCRIPTION
**Pull request by mbolivar**
West sign was broken when build.dir-fmt was added.

Fix that and add some improvements while we are here.

This change breaks backwards compatibility; the default output
locations are now in the build directory.

### CODE REVIEW

**Review by carlescufi — APPROVED** [review]
LGTM, but have not tested it

**Comment by mbolivar** [comment]
Rebased and updated documentation also.

File: doc/guides/west/sign.rst [hunk 1]
```diff
@@ -35,25 +35,33 @@ Some additional notes follow. See ``west sign -h`` for detailed help.
...
 
 .. code-block:: console
 
    west build -b YOUR_BOARD samples/hello_world -- -DCONFIG_BOOTLOADER_MCUBOOT=y
    west sign -t imgtool -- --key YOUR_SIGNING_KEY.pem
-   west flash --hex-file zephyr.signed.hex
+   west flash --hex-file build/zephyr/zephyr.signed.hex
 
-Availability of a hex file depends on your build configuration.
+The availability of a hex file, and whether west flash uses it to flash,
```
Comment (dbkinder) [inline doc/guides/west/sign.rst]:
```suggestion
The availability of a hex file, and whether ``west flash`` uses it to flash,
```

**Review by dbkinder — CHANGES_REQUESTED** [review]
one nit fix, otherwise LGTM

File: doc/guides/west/sign.rst (hunk 1 above)
Comment (mbolivar) [reply doc/guides/west/sign.rst]: Done!

**Review by dbkinder — APPROVED** [review]
doc changes LGTM, thanks!

**Review by mnkp — APPROVED** [review]
@mbolivar Sorry for the late review. I've tested the PR on `efr32_slwstk6061a` board with `jlink` runner. The signed images are generated as expected. I could successfully run the hello_world sample app after flashing the `zephyr.signed.bin` image. Application flashed via the `zephyr.signed.hex` image doesn't run but it was also the case before. I didn't have time to debug the issue yet but `west sign` is unlikely the cause. Another user complained about similar issue in "[Zephyr-devel] MCUBoot can't find bootable image (nRF52)". He was running `imgtool` "by hand".

**Comment by mbolivar** [comment]
@mnkp great, thanks so much for your review and testing. Sorry to hear about the hex issues you're facing, though. I wonder what's going on -- have you asked the mcuboot developers?

**Comment by mnkp** [comment]
No, I didn't have time to look into the .hex issue, neither contact MCUBoot developers. I'll try to do it.

Have you try to flash .hex file yourself? If it generally doesn't work that's a bit of an issue, since few Zephyr runners correctly flash .bin images.

**Comment by mbolivar** [comment]
Yeah, I always flash hex files, since it's more convenient because they have embedded addresses.

**Pull request merged**